app/assets/javascripts/discourse/app/routes/tag-show.js

```diff
@@ -23,100 +23,112 @@ export default DiscourseRoute.extend(FilterModeMixin, {
 
   queryParams,
 
-  renderTemplate() {
+  controllerName: "tags.show",
+  templateName: "tags.show",
+
+  beforeModel() {
     const controller = this.controllerFor("tag.show");
-    this.render("tags.show", { controller });
+    controller.setProperties({
+      loading: true,
+      showInfo: false,
+    });
   },
 
-  model(params) {
+  async model(params, transition) {
     const tag = this.store.createRecord("tag", {
       id: escapeExpression(params.tag_id),
     });
+
+    let additionalTags;
+
     if (params.additional_tags) {
-      this.set(
-        "additionalTags",
-        params.additional_tags.split("/").map((t) => {
-          return this.store.createRecord("tag", {
-            id: escapeExpression(t),
-          }).id;
-        })
-      );
-    } else {
-      this.set("additionalTags", null);
+      additionalTags = params.additional_tags.split("/").map((t) => {
+        return this.store.createRecord("tag", {
+          id: escapeExpression(t),
+        }).id;
+      });
     }
 
-    this.set("filterType", this.navMode.split("/")[0]);
+    const filterType = this.navMode.split("/")[0];
 
-    this.set("categorySlugPathWithID", params.category_slug_path_with_id);
-
-    if (tag && tag.get("id") !== "none" && this.currentUser) {
+    let tagNotification;
+    if (tag && tag.id !== "none" && this.currentUser) {
```

Comment:
```suggestion
    if (tag?.id !== "none" && this.currentUser) {
```

Reply:
This would change the logic - so I don't think we can simplify it.

<img width="228" alt="Screenshot 2022-03-18 at 14 57 09" src="https://user-images.githubusercontent.com/6270921/159026957-85765e1c-2feb-4484-bf25-99a8bfb067cc.png">

Reply:
rule of thumb: don't mix `?.` and `!==` 😃